pkg/auth/oauth2authenticator.go

```diff
@@ -359,47 +408,35 @@ func (o *oAuth2Authenticator) authenticateWithAuthorizationCode() error {
 		// launch browser
 		go o.openBrowserFunc(authCodeUrl)
 
-		timedOut := false
-		timer := time.AfterFunc(TIMEOUT_SECONDS, func() {
-			timedOut = true
+		// Handle context cancellation
+		var timedOut atomic.Bool
+		var canceled atomic.Bool
+		go func() {
+			<-ctx.Done()
+			switch err := ctx.Err(); {
+			case errors.Is(err, context.DeadlineExceeded):
+				timedOut.Store(true)
+			case errors.Is(err, context.Canceled):
+				canceled.Store(true)
+			}
```

Comment:
why isn't that done in the error handling below? That way we would have error handling in one place.